azurerm/resource_arm_function_app.go

```diff
@@ -469,19 +484,45 @@ func getBasicFunctionAppAppSettings(d *schema.ResourceData) []web.NameValuePair
 	functionVersion := d.Get("version").(string)
 	contentShare := d.Get("name").(string) + "-content"
 
-	return []web.NameValuePair{
+	basicSettings := []web.NameValuePair{
 		{Name: &dashboardPropName, Value: &storageConnection},
 		{Name: &storagePropName, Value: &storageConnection},
 		{Name: &functionVersionPropName, Value: &functionVersion},
+	}
+
+	consumptionSettings := []web.NameValuePair{
 		{Name: &contentSharePropName, Value: &contentShare},
 		{Name: &contentFileConnStringPropName, Value: &storageConnection},
 	}
+
+	// If the application plan is dynamic (consumption), we do not want to include WEBSITE_CONTENT components
+	if strings.EqualFold(appServiceTier, "dynamic") {
+		return basicSettings
+	}
+	return append(basicSettings, consumptionSettings...)
+}
+
+func getFunctionAppServiceTier(ctx context.Context, appServicePlanId string, meta interface{}) (string, error) {
+	id, err := parseAzureResourceID(appServicePlanId)
+	if err != nil {
+		return "", fmt.Errorf("[ERROR] Unable to parse App Service Plan ID '%s': %+v", appServicePlanId, err)
+	}
+
+	log.Printf("[DEBUG] Retrieving App Server Plan %s", id.Path["serverfarms"])
+
+	appServicePlansClient := meta.(*ArmClient).appServicePlansClient
+	appServicePlan, err := appServicePlansClient.Get(ctx, id.ResourceGroup, id.Path["serverfarms"])
+	if err != nil {
+		return "", fmt.Errorf("[ERROR] Could not retrieve App Service Plan ID '%s': %+v", appServicePlanId, err)
+	}
+
+	return *appServicePlan.Sku.Tier, nil
```

Comment:
there's a crash here if `appServicePlan.Sku` or `appServicePlan.Sku.Tier` is nil (which happens for older resources); can we make this:

```
if sku := appServicePlan.Sku; sku != nil {
 if tier := sku.Tier; tier != nil {
    return *tier, nil
  }
}
return fmt.Errorf("No `sku` block was returned for App Service Plan %q (Resource Group %q)", resourceGroup, name)
```